Initiated tests for `reduce_data()`

FriedaRosa opened the pull request:
The following tests have been added for the function `reduce_data()`.
Different modifications of the `example_data` were produced to account for edge cases. Within each test there is always a function to test for the correct output structure*.

## Edge cases
1. if one taxon is all-zero in the community data
2. if all taxa are all-zero in the community data
3. if one taxon has NAs in the community data
4. if there is one level with NA in the community data
5. if there is one level with zeros in the community data
6. if all levels are zero in the community data

## Tests
### General functionality:
- Verification that taxa are reduced correctly 
- Verification that levels are reduced correctly
- Verification that both are reduced correctly at the same time
- Verification that `check_taxa = FALSE` and `check_levels = FALSE` does not manipulate the data

### Edge case scenarios:
- Verification of scenario [1]
- Verification of scenario [2]
- Verification of scenario [3]
- Verification of scenario [4]
- Verification of scenario [5]
- Verification of scenario [6]

## Comments:
!* All of the tests for correct output structure fail as long as there is the bug (#44 ) in `extract_data()` that leads to no `sample_id` information in the age uncertainty matrix.

FriedaRosa committed:
Create test-reduce_data.R

So far, the tests for the output data structure all fail because of a bug in extract_data() (solution proposed in #44).

FriedaRosa committed:
Removed redundant tests

OndrejMottl committed:
test-reduce_data: styler

OndrejMottl reviewed (changes requested):
Great!!! This one is super cool!
My small request is to also test that the input assertion is working. In other words, test that all input arguments throw an error is passed wrong data.

FriedaRosa committed:
split up tests and added input validation

FriedaRosa committed:
Improvements towards better structure and higher independence of individual unit tests

- no more code outside of unit tests
- only one (or few) expections per test
- input validation more thorough including more scenarios

FriedaRosa committed:
Removed redundant tests and merged others

Trying to minimize cluttering the test files with redundant tests while keeping high coverage across use cases, edge cases and combinations of parameters for each function argument.

FriedaRosa committed:
fixed actual error messages for testing

FriedaRosa committed:
final cleanup

FriedaRosa committed:
Update test-reduce_data.R

Corrected style, removed uneccesary tests, corrected expectations

OndrejMottl committed:
styling
